app/jobs/fetch_oa_metadata_job.rb

```diff
@@ -15,6 +15,7 @@ def perform(publication_id)
       publication.open_access_button_last_checked_at = Time.zone.now
     end
 
+    unpaywall_response.oa_status.present? ? publication.update_oa_status_from_unpaywall(unpaywall_response) : publication.open_access_status = 'unknown'
```

Comment:
Yeah, you could probably get rid of the OAB stuff here.  I guess there's the rare chance something is in OAB but not Unpaywall, in which case I think we'd want to exclude those from this workflow, though. 

I don't think we have anything built in RMD to check ScholarSphere for a single record.  The ScholarSphere import just grabs everything in ScholarSphere and sees if there's a matching DOI in RMD.  I'm pretty sure there is a DOI endpoint in ScholarSphere's API, so something could be built.

I agree we don't need all the metadata from Unpaywall for this job, so if we wanted to simplify things, we could just get whatever is necessary to validate and the open access status.  Then, leave the rest to the other importers.

Reply:
Yeah, here's how I'd prioritize the changes that I suggested:

**High priority:**

- I think that we must check for ScholarSphere open access locations here one way or another.

**Low priority:**

- It would simplify things to remove the OAB checking/importing here since it's no longer needed, but it doesn't _hurt_ anything functionality-wise to leave it in. However, if we leave it in, then I'd say that it's high priority to ensure that we're not creating duplicate open access location records.
- Likewise, I think that the Unpaywall checking/importing is probably working correctly as is, but it looks to me like it could be made simpler either by only importing the OA status here or by just importing everything that we ultimately want from Unpaywall (OA status, OA locations, etc.) for these publications in the same way that we do elsewhere.